test/test_routes.py

```diff
@@ -45,6 +45,44 @@ def test_predict_route(self):
         output = dict(response.json())
         self.assertEqual(output["data"], ["testtest"])
 
+    def test_named_predict_route(self):
+        with Blocks() as demo:
+            i = Textbox()
+            o = Textbox()
+            i.change(lambda x: x + "1", i, o, api_name="p")
+            i.change(lambda x: x + "2", i, o, api_name="q")
+
+        app, _, _ = demo.launch(prevent_thread_lock=True)
+        client = TestClient(app)
+        response = client.post("/api/p/", json={"data": ["test"]})
+        assert response.status_code == 200
+        output = dict(response.json())
+        assert output["data"] == ["test1"]
+
+        response = client.post("/api/q/", json={"data": ["test"]})
+        assert response.status_code == 200
+        output = dict(response.json())
+        assert output["data"] == ["test2"]
+
+    def test_same_named_predict_route(self):
```

Comment:
Might be nice to test a really pathological case like
```python
i.change(lambda x: x + "0", i, o, api_name="p")
i.change(lambda x: x + "1", i, o, api_name="p")
i.change(lambda x: x + "2", i, o, api_name="p_1")
```

To test the renaming scheme handles collisions after one of the original names has been changed.

Reply:
Ah yes, good point!! That's a fun one, the "p_1" will get renamed to "p_2" so maybe I should throw a warning when "api_name" is changed